.gitlab-ci.yml

```diff
@@ -1,121 +1,207 @@
+## TODO: Modularize the gitlab yaml to reuse templates
+
+## Define the stages & order of execution
+
 stages:
-  - cluster-setup
-  - infra-setup
-  - stateful-apps-deploy
-  - litmus-chaos
-  - cleanup
-  - deploy
+  - CLUSTER-SETUP
+  - PROVIDER-INFRA-SETUP
+  - STATEFUL-APP-DEPLOY
+  - APP-FUNC-TEST
+  - APP-CHAOS-TEST
+  - CLUSTER-CLEANUP
+
+## Setup the kubernetes cluster
+
 azure-cluster:
   image: atulabhi/kops:v8
-  stage: cluster-setup
+  stage: CLUSTER-SETUP 
   script: 
-    - chmod 755 ./script/azure
     - ./script/azure
-
   artifacts:
     paths:
       - azoebs_aks_name/
 
-openebs-deploy:
+## Setup OpenEBS control plane
+
+openebs-azure-deploy:
   image: atulabhi/kops:v8
-  stage: infra-setup
+  stage: PROVIDER-INFRA-SETUP
   dependencies:
     - azure-cluster
   script: 
-   - echo "deploying OpenEBS on Azure"
-   - ls
-   - chmod 775 ./script/infra-setup
-   - ./script/infra-setup
-  
+   - ./script/provider/infra-setup
```

Comment:
Is this script working without `chmod 775 ./script/infra-setup` ?

Reply:
Granted permissions for all the scripts already.

Reply:
:+1: